packages/web-service/src/pages/eligibility/eligibility.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Route handlers for the eligibility check flow
+ */
+import { yesNoPage, isYes } from '../common/yes-no.js'
+import { checkAnswersPage } from '../common/check-answers.js'
+import { eligibilityURIs } from '../../uris.js'
+import pageRoute from '../../routes/page-route.js'
+// The pages in the flow
+const {
+  LANDOWNER, LANDOWNER_PERMISSION, CONSENT, CONSENT_GRANTED,
+  NOT_ELIGIBLE_LANDOWNER, NOT_ELIGIBLE_PROJECT, ELIGIBILITY_CHECK, ELIGIBLE
+} = eligibilityURIs
+
+const IS_OWNER_OF_LAND = 'isOwnerOfLand'
+const HAS_LANDOWNER_PERMISSION = 'hasLandOwnerPermission'
+const PERMISSION_REQUIRED = 'permissionsRequired'
+const PERMISSION_GRANTED = 'permissionsGranted'
+
+// Helper to operate on the eligibility section of the journey cache
+export const eligibilityHelper = async (request, operator) => {
+  const journeyData = await request.cache().getData() || { eligibility: {} }
+  const { eligibility } = journeyData
+  operator(request, eligibility)
+  await request.cache().setData(journeyData)
+}
+
+// A state machine to determine the next page
+export const eligibilityCompletion = async request => {
+  const journeyData = await request.cache().getData() || { eligibility: {} }
+  const { eligibility } = journeyData
+  const grantedCompletionSection = e => {
+    if (e[PERMISSION_GRANTED] === undefined) {
+      return CONSENT_GRANTED.uri
+    } else if (!e[PERMISSION_GRANTED]) {
+      return NOT_ELIGIBLE_PROJECT.uri
+    } else {
+      return ELIGIBILITY_CHECK.uri
+    }
+  }
+
+  const consentCompletionSection = e => {
+    if (e[PERMISSION_REQUIRED] === undefined) {
+      return CONSENT.uri
+    } else if (!e[PERMISSION_REQUIRED]) {
+      return ELIGIBILITY_CHECK.uri
+    } else {
+      return grantedCompletionSection(e)
+    }
+  }
+
+  const permissionsCompletionSection = e => {
+    if (e[HAS_LANDOWNER_PERMISSION] === undefined) {
+      return LANDOWNER_PERMISSION.uri
+    } else if (!e[HAS_LANDOWNER_PERMISSION]) {
+      return NOT_ELIGIBLE_LANDOWNER.uri
+    } else {
+      return consentCompletionSection(eligibility)
+    }
+  }
+
+  // Must be set
+  if (eligibility[IS_OWNER_OF_LAND] === undefined) {
+    return LANDOWNER.uri
+  } else if (eligibility[IS_OWNER_OF_LAND]) {
+    // Owner
+    return consentCompletionSection(eligibility)
+  } else {
+    // Not the owner
+    return permissionsCompletionSection(eligibility)
+  }
+}
+
+/**************************************************************
+ * Are you the landowner?
+ **************************************************************/
+export const landOwnerSetData = request =>
+  eligibilityHelper(request, (r, e) => {
```

Comment:
The anonymous `(r, e) => ...` function already has access to the `request` object from its closure so doesn't need to have it passed back in again and only needs to be defined as `e => ...`

Reply:
See above